### ISSUE TITLE
linus master (5.0-rc1) x86 broken by asm goto

### ISSUE THREAD

**Issue by shenki** [ClangBuiltLinux/continuous-integration#109]
https://travis-ci.com/ClangBuiltLinux/continuous-integration/jobs/168455285#L643

```
In file included from arch/x86/kernel/asm-offsets.c:9:
In file included from ./include/linux/crypto.h:26:
In file included from ./include/linux/uaccess.h:11:
In file included from ./arch/x86/include/asm/uaccess.h:692:
./arch/x86/include/asm/uaccess_64.h:132:3: error: 'asm goto' constructs are not supported yet
                __put_user_asm(*(u8 *)src, (u8 __user *)dst,
                ^
./arch/x86/include/asm/uaccess.h:477:11: note: expanded from macro '__put_user_asm'
        retval = __put_user_failed(x, addr, itype, rtype, ltype, errret);       \
                 ^
./arch/x86/include/asm/uaccess.h:472:3: note: expanded from macro '__put_user_failed'
                __put_user_goto(x,addr,itype,rtype,ltype,__puflab);     \
                ^
./arch/x86/include/asm/uaccess.h:463:2: note: expanded from macro '__put_user_goto'
        asm_volatile_goto("\n"                                          \
        ^
./include/linux/compiler_types.h:188:37: note: expanded from macro 'asm_volatile_goto'
#define asm_volatile_goto(x...) asm goto(x)
                                    ^
```

The commit [e9666d10a567](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=e9666d10a5677a494260d60d1fa0b73cc7646eb3 ) ("jump_label: move 'asm goto' support test to Kconfig") looks suspicious. Here's what it does when using clang:

```
$ scripts/gcc-goto.sh clang-8
<stdin>:13:6: error: 'asm goto' constructs are not supported yet
        asm goto ("" :::: entry);
            ^
1 error generated.
$ echo $?
1
```
With GCC:
```
$ scripts/gcc-goto.sh clang-8
<stdin>:13:6: error: 'asm goto' constructs are not supported yet
        asm goto ("" :::: entry);
            ^
1 error generated.
$ echo $?
1
```

**Comment by nathanchance** [comment]
It's actually that x86 is actually using asm gotos outside of HAVE_JUMP_LABEL. We'll need to revert the two commits that use them to keep building the kernel with Clang. I have the revert queued up, I was waiting for those commits to hit next.

https://github.com/ClangBuiltLinux/linux/issues/6#issuecomment-451676502

**Comment by shenki** [comment]
Those asm-goto using commits weren't in next? That's poor form from the maintainer responsible.

Thanks for the fixes.

**Comment by nathanchance** [comment]
Unfortunately, Linus authored them and applied them himself as a part of some uaccess cleanup so there was no maintainer involved. Alas. We knew it was coming once asm goto support was forced for x86, we've just been fortunate that it hasn't been actually been used. Hopefully it stops here for now.

**Comment by nathanchance** [comment]
Fixed (for now) with #110.

Closing since https://github.com/ClangBuiltLinux/linux/issues/6 is the master issue for asm goto.
